File: event-sourcing/csharp/src/EventSourcing/Exceptions.cs
namespace EventSourcing;

public class AccountNotOpenException : Exception
{
    public AccountNotOpenException(string message) : base(message) { }
}

public class AccountClosedException : Exception
{
    public AccountClosedException(string message) : base(message) { }
}

public class InsufficientFundsException : Exception
{
    public InsufficientFundsException(string message) : base(message) { }
}

public class InvalidAmountException : Exception
{
    public InvalidAmountException(string message) : base(message) { }
}

public class NonZeroBalanceException : Exception
{
    public NonZeroBalanceException(string message) : base(message) { }
}

File: event-sourcing/typescript/src/Exceptions.ts
export class AccountNotOpenException extends Error {
  constructor(message: string) { super(message); this.name = 'AccountNotOpenException'; }
}

export class AccountClosedException extends Error {
  constructor(message: string) { super(message); this.name = 'AccountClosedException'; }
}

export class InsufficientFundsException extends Error {
  constructor(message: string) { super(message); this.name = 'InsufficientFundsException'; }
}

export class InvalidAmountException extends Error {
  constructor(message: string) { super(message); this.name = 'InvalidAmountException'; }
}

export class NonZeroBalanceException extends Error {
  constructor(message: string) { super(message); this.name = 'NonZeroBalanceException'; }
}
